dev/release/verify-release-candidate.sh

```diff
@@ -378,22 +342,253 @@ test_csharp() {
+      $python -m virtualenv ${virtualenv}
+    fi
+    # Activate the environment
+    source "${virtualenv}/bin/activate"
+    # Install dependencies
+    if [ $# -gt 0 ]; then
+      pip install $@
+    fi
+  fi
+}
+
+setup_go() {
+  if [ "${USE_CONDA}" -eq 0 ]; then
+    install_go
+  fi
+}
+
+setup_nodejs() {
+  if [ "${USE_CONDA}" -eq 0 ]; then
+    install_nodejs
+  fi
+}
+
+test_package_java() {
+  # Build and test Java (Requires newer Maven -- I used 3.3.9)
+  setup_conda maven
+
+  pushd java
+  mvn test
+  mvn package
   popd
 }
 
-# Build and test Python
+test_and_install_cpp() {
+  # Build and test C++
+  setup_virtualenv numpy
+  setup_conda \
+    --file ci/conda_env_unix.txt \
+    --file ci/conda_env_cpp.txt \
+    --file ci/conda_env_gandiva.txt \
+    ncurses \
+    numpy \
+    sqlite \
+    compilers
+
+  if [ "${USE_CONDA}" -gt 0 ]; then
+    DEFAULT_DEPENDENCY_SOURCE="CONDA"
+  else
+    DEFAULT_DEPENDENCY_SOURCE="AUTO"
+  fi
+
+  mkdir -p $ARROW_TMPDIR/build
+  pushd $ARROW_TMPDIR/build
+
+  if [ ! -z "$CMAKE_GENERATOR" ]; then
+    ARROW_CMAKE_OPTIONS="${ARROW_CMAKE_OPTIONS:-} -G Ninja"
+  fi
+
+  cmake \
+    -DARROW_BOOST_USE_SHARED=ON \
+    -DARROW_BUILD_INTEGRATION=ON \
+    -DARROW_BUILD_TESTS=ON \
+    -DARROW_CUDA=${ARROW_CUDA} \
+    -DARROW_DATASET=ON \
+    -DARROW_DEPENDENCY_SOURCE=${ARROW_DEPENDENCY_SOURCE:-$DEFAULT_DEPENDENCY_SOURCE} \
+    -DARROW_FLIGHT=${ARROW_FLIGHT} \
+    -DARROW_GANDIVA=${ARROW_GANDIVA} \
+    -DARROW_HDFS=ON \
+    -DARROW_ORC=ON \
+    -DARROW_PARQUET=ON \
+    -DARROW_PLASMA=${ARROW_PLASMA} \
+    -DARROW_PYTHON=ON \
+    -DARROW_S3=${ARROW_S3} \
+    -DARROW_USE_CCACHE=${ARROW_USE_CCACHE:-OFF} \
+    -DARROW_VERBOSE_THIRDPARTY_BUILD=ON \
+    -DARROW_WITH_BROTLI=ON \
+    -DARROW_WITH_BZ2=ON \
+    -DARROW_WITH_LZ4=ON \
+    -DARROW_WITH_SNAPPY=ON \
+    -DARROW_WITH_ZLIB=ON \
+    -DARROW_WITH_ZSTD=ON \
+    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-release} \
+    -DCMAKE_INSTALL_LIBDIR=lib \
+    -DCMAKE_INSTALL_PREFIX=$ARROW_HOME \
+    -DCMAKE_UNITY_BUILD=${CMAKE_UNITY_BUILD:-OFF} \
+    -DGTest_SOURCE=BUNDLED \
+    -DPARQUET_REQUIRE_ENCRYPTION=ON \
+    ${ARROW_CMAKE_OPTIONS:-} \
+    ${ARROW_SOURCE_DIR}/cpp
+  cmake --build . --target install
+
+  # TODO: ARROW-5036: plasma-serialization_tests broken
+  # TODO: ARROW-5054: libgtest.so link failure in flight-server-test
+  LD_LIBRARY_PATH=$PWD/release:$LD_LIBRARY_PATH ctest \
+    --exclude-regex "plasma-serialization_tests" \
+    -j$NPROC \
+    --output-on-failure \
+    -L unittest
+
+  popd
+}
 
 test_python() {
-  pushd python
+  # Build and test Python
+  setup_virtualenv cython numpy setuptools_scm setuptools
+  setup_conda --file ci/conda_env_python.txt
 
   export PYARROW_PARALLEL=$NPROC
-
   export PYARROW_WITH_DATASET=1
+  export PYARROW_WITH_HDFS=1
+  export PYARROW_WITH_ORC=1
   export PYARROW_WITH_PARQUET=1
   export PYARROW_WITH_PLASMA=1
+  if [ "${ARROW_S3}" = "ON" ]; then
```

Comment:
Can you keep these entries alphabetically ordered?

Reply:
Updated.